### PR TITLE
fix: add getBlock Connection method

### PR DESCRIPTION
**Pull request by CriesofCarrots**
Will add issue to update getBlock test on upstream updates

### CODE REVIEW

**Comment by CriesofCarrots** [comment]
CI failure is because this hasn't propagated to our docker image yet: https://github.com/solana-labs/solana/pull/6889

File: src/connection.js [hunk 1]
```diff
@@ -904,6 +916,23 @@ export class Connection {
     return res.result;
   }
 
+  /**
+   * Fetch a list of Transactions and transaction statuses from the cluster
+   */
+  async getBlock(
+    slot: number,
+  ): Promise<Array<[Transaction, SignatureSuccess | TransactionError | null]>> {
```
Comment (jstarry) [inline src/connection.js]: esdoc is complaining about this type ^

It looks like it's supposed to be: `[Transaction, SignatureSuccess] | ...` just missing a closing bracket

Comment (CriesofCarrots) [reply src/connection.js]: The closing bracket is after the `null`...
I tried putting the whole OR expression (the second value in the tuple) in parens, but prettier removed them

Comment (CriesofCarrots) [reply src/connection.js]: Maybe I have to be explicit?
`[Transaction, SignatureSuccess] | [Transaction, TransactionError] ...`

Comment (CriesofCarrots) [reply src/connection.js]: Yes, the explicit tuples approach works.
Thanks, @jstarry !

**Pull request merged**